app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rposts/RPostViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.sharewhatyoucanproject.rposts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.GeoPoint
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlin.collections.ArrayList
+
+class RPostViewModel(
+    private val db: FirebaseFirestore,
+) : ViewModel() {
+    var arrayList: ArrayList<PostModel> = mutableListOf<PostModel>() as ArrayList<PostModel>
+    private val _taskResult = MutableLiveData<TaskResult>()
+    val taskResult: LiveData<TaskResult> = _taskResult
+
+    fun getlist() {
+        db.collection("posts")
+            .whereEqualTo("status", 0)
```

Comment:
what does this mean? Is there a way to make it more descriptive?

Reply:
Yeah sure i will change it